command/agent/sink/file/file_sink_test.go

```diff
@@ -80,3 +80,68 @@ func TestFileSink(t *testing.T) {
 		t.Fatalf("expected %s, got %s", uuidStr, string(fileBytes))
 	}
 }
+
+func testFileSinkMode(t *testing.T, log hclog.Logger) (*sink.SinkConfig, string) {
+	tmpDir, err := ioutil.TempDir("", fmt.Sprintf("%s.", fileServerTestDir))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	path := filepath.Join(tmpDir, "token")
+
+	config := &sink.SinkConfig{
+		Logger: log.Named("sink.file"),
+		Config: map[string]interface{}{
+			"path": path,
+			"mode": 0644,
+		},
+	}
+
+	s, err := NewFileSink(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	config.Sink = s
+
+	return config, tmpDir
+}
+
+func TestFileSinkMode(t *testing.T) {
+	log := logging.NewVaultLogger(hclog.Trace)
+
+	fs, tmpDir := testFileSinkMode(t, log)
+	defer os.RemoveAll(tmpDir)
+
+	path := filepath.Join(tmpDir, "token")
+
+	uuidStr, _ := uuid.GenerateUUID()
+	if err := fs.WriteToken(uuidStr); err != nil {
+		t.Fatal(err)
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fi, err := file.Stat()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if fi.Mode() != os.FileMode(0644) {
+		t.Fatalf("wrong file mode was detected at %s", path)
+	}
+	err = file.Close()
```

Comment:
Should this be in a `defer` right after the file is opened? I suppose if the test kept stopping at line 132 on a CI server, it might be possible for it to eventually run out of available file handles. Definitely an edge case, though.